src/crimson/os/seastore/segment_cleaner.h

```diff
@@ -641,7 +642,9 @@ class SegmentCleaner : public SegmentProvider {
   void set_journal_head(journal_seq_t head) {
     assert(journal_head == JOURNAL_SEQ_NULL || head >= journal_head);
     journal_head = head;
-    segments.update_written_to(head.offset);
+    if (head.offset.get_addr_type() == addr_types_t::SEGMENT) {
```

Comment:
This should also be an assert.  Once you add support, it will be entirely fine to call set_journal_head on addresses for the CBJournal.